compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3371,6 +3371,244 @@ static void arrayCopyDefault(TR::Node* node, uint8_t elementSize, TR::Register*
       }
    }
 
+/*
+ * Generate a store instruction
+ * Parameters:
+ * node: the tree node
+ * addressReg: the base register for the destination mem address
+ * index: the index for the destination mem address
+ * valueReg: the value that needs be stored
+ * size: the size of valueReg
+ * cg: the code generator
+ */
+static void generateStoreInstructions(TR::Node* node, TR::Register* addressReg, int32_t index, TR::Register* valueReg, uint8_t size,  TR::CodeGenerator* cg)
+   {
+   TR_X86OpCodes storeOpcode;
+   if (valueReg->getKind() == TR_FPR)
+      {
+      switch (size)
+         {
+         case 4:
+            storeOpcode = MOVDMemReg;
+            break;
+         case 8:
+            storeOpcode = MOVQMemReg;
+            break;
+         case 16:
+            storeOpcode = MOVDQUMemReg;
+            break;
+         }
+      }
+   else 
```

Comment:
There are vector registers as well as GPRs - for safety make the GPR case explict and add an else and assert in case you don't get either of the two handled kinds of registers. Makes future extension easier to handle too.